eng/pipelines/templates/steps/run_pyright.yml

```diff
@@ -0,0 +1,49 @@
+parameters:
+  BuildTargetingString: 'azure-*'
+  ServiceDirectory: ''
+  TestMarkArgument: ''
+  EnvVars: {}
+
+# The variable TargetingString is set by template `eng/pipelines/templates/steps/targeting-string-resolve.yml`. This template is invoked from yml files:
+#     eng/pipelines/templates/jobs/ci.tests.yml
+#     eng/pipelines/templates/jobs/ci.yml
+#     eng/pipelines/templates/jobs/live.test.yml
+
+# Please use `$(TargetingString)` to refer to the python packages glob string. This was previously `${{ parameters.BuildTargetingString }}`.
+steps:
+  - task: UsePythonVersion@0
+    displayName: 'Use Python 3.10'
+    inputs:
+     versionSpec: '3.10'
+    condition: and(succeededOrFailed(), ne(variables['Skip.Pyright'],'true'))
```

Comment:
Is the intent that one would disable pyright for one-off pipeline runs, or that a service could exclude it for all pipelines? If the latter, I think a parameter that gets passed down to this template from `ci.yml` would be a better approach, otherwise we'd have to register variables in the pipelines which aren't as easily inspectable or changeable en masse.

Reply:
Intent is that pyright could be excluded for all pipelines for a service. I don't imagine people will disable for one-off runs. This matches what we currently do with mypy, our other type checker, and pylint. Are you thinking that this should be updated across all static / linting checks?

Reply:
Discussed offline. The tox harness will be extended to honor the common exclusion list in this PR and won't invoke a tox environment if it's unsupported. This will help save build time.